### PR TITLE
Assign number of ports after initialization is complete 

### PR DESCRIPTION
**Pull request by tokle**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by tokle** [comment]
@bjorncs, @andreer fyi

File: config-model/src/main/java/com/yahoo/vespa/model/container/Container.java [hunk 1]
```diff
@@ -89,19 +87,24 @@ protected Container(AbstractConfigProducer parent, String name, boolean retired,
...
-            numHttpServerPorts = getHttp().getHttpServer().getConnectorFactories().size();
         }
         addBuiltinHandlers();
 
         addChild(new SimpleComponent("com.yahoo.container.jdisc.ConfiguredApplication$ApplicationContext"));
     }
 
+    protected int getNumHttpServerPorts() {
+        if (getHttp() == null) {
+            // TODO Vespa 8: set to 1. The second (health) port has not been used since Vespa 6 or earlier.
```
Comment (andreer) [inline config-model/src/main/java/com/yahoo/vespa/model/container/Container.java]: The comment stops making sense with this change.

File: config-model/src/main/java/com/yahoo/vespa/model/container/Container.java [hunk 2]
```diff
@@ -89,19 +87,24 @@ protected Container(AbstractConfigProducer parent, String name, boolean retired,
...
+            return defaultHttpServer.getConnectorFactories().size()+1;
```
Comment (arnej27959) [inline config-model/src/main/java/com/yahoo/vespa/model/container/Container.java]: adding +1 is only correct before initDefaultJettyConnector has added one, but this method needs to work both before and after that :-(

**Comment by tokle** [comment]
Close in favor of #10500

**Pull request closed without merging**